compiler.pri

```diff
@@ -227,39 +227,61 @@ unix {
 		QMAKE_OBJECTIVE_CXXFLAGS *= -O3 -march=native -ffast-math -ftree-vectorize -fprofile-use
 	}
 
-	CONFIG(c++11) {
-		# Qt 5 will pass the expected compiler flags
-		# needed for C++11 mode when CONFIG includes c++11.
-		# But Qt 4 won't, so we add it manually.
-		lessThan(QT_MAJOR_VERSION, 5) {
-			QMAKE_CXXFLAGS *= -std=c++11
+	# First, auto-populate CONFIG with the latest C++ standard
+	# that Qt has detected support for.
+	!CONFIG(c++03):!CONFIG(c++11):!CONFIG(c++14):!CONFIG(c++1z) {
+		contains(QT_CONFIG, c++11): CONFIG *= c++11
+		contains(QT_CONFIG, c++14): CONFIG *= c++14
+		contains(QT_CONFIG, c++1z): CONFIG *= c++1z
+	}
+	# If no C++ standard was auto-detected, choose C++03.
+	!CONFIG(c++03):!CONFIG(c++11):!CONFIG(c++14):!CONFIG(c++1z) {
+		CONFIG += c++03
+	}
+
+	# Qt 4, as opposed to Qt 5, doesn't automatically
+	# populate QMAKE_CXXFLAGS and QMAKE_LFLAGS with the
+	# appproiate -std=c++XX flags. So, for Qt 4, we'll
+	# have to set them up manually.
+	lessThan(QT_MAJOR_VERSION, 5) {
+		CONFIG(c++11) {
+			QMAKE_CXXFLAGS += -std=c++11
+			QMAKE_LFLAGS += -std=c++11
+		}
+		CONFIG(c++14) {
+			QMAKE_CXXFLAGS += -std=c++14
+			QMAKE_LFLAGS += -std=c++14
+		}
+		CONFIG(c++1z) {
+			QMAKE_CXXFLAGS += -std=c++1z
+			QMAKE_LFLAGS += -std=c++1z
 		}
```

Comment:
This section won't work if more than one `CONFIG` option is set.  g++ doesn't like having multiple `-std=...` lines passed to it.

If QT is compiled with support for more than one of these, then more than one `CONFIG` option is set.